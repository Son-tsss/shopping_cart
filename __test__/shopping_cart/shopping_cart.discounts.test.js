import ShoppingCart from '../../src/shopping_cart/shopping_cart_actions'

it('should apply discount to the existing item', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.applyDiscountToItem(0.5, newItem.id)

    const real = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('applying discount to not existing item should change nothing', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.applyDiscountToItem(0.5, "i3")

    const real = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('should apply discount to cart', ()=>{
    const shoppingCart = new ShoppingCart

    shoppingCart.applyDiscountToCart(0.5)

    const real = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})
