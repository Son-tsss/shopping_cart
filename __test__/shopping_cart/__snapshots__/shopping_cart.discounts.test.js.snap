// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applying discount to not existing item should change nothing 1`] = `
Object {
  "currentState": Object {
    "discount": 1,
    "items": Array [
      Object {
        "count": 1,
        "discount": 1,
        "id": "i1",
        "price": 100,
      },
    ],
  },
  "futureStates": Array [],
  "previousStates": Array [
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [
        Object {
          "count": 1,
          "discount": 1,
          "id": "i1",
          "price": 100,
        },
      ],
    },
  ],
}
`;

exports[`should apply discount to cart 1`] = `
Object {
  "currentState": Object {
    "discount": 0.5,
    "items": Array [],
  },
  "futureStates": Array [],
  "previousStates": Array [
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [],
    },
  ],
}
`;

exports[`should apply discount to the existing item 1`] = `
Object {
  "currentState": Object {
    "discount": 1,
    "items": Array [
      Object {
        "count": 1,
        "discount": 0.5,
        "id": "i1",
        "price": 100,
      },
    ],
  },
  "futureStates": Array [],
  "previousStates": Array [
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [
        Object {
          "count": 1,
          "discount": 1,
          "id": "i1",
          "price": 100,
        },
      ],
    },
  ],
}
`;
